luaui/Widgets/cmd_mex_snap.lua

```diff
@@ -33,36 +70,33 @@ local spGetUnitMetalExtraction = Spring.GetUnitMetalExtraction
 local spGiveOrder = Spring.GiveOrder
 local math_pi = math.pi
 local preGamestartPlayer = Spring.GetGameFrame() == 0 and not Spring.GetSpectatingState()
-local spGetUnitPosition = Spring.GetUnitPosition
-local spIsUnitAllied = Spring.IsUnitAllied
-
 
 local activeCmdID, bx, by, bz, bface
 local unitshape
 local curPosition
 
+-- These use api_resource_spot_{finder,builder} and are assigned at initialization
+local isMexConstructor = function()
+	return false
+end
 local isMex = {}
+local metalSpotsList = {}
+local GetBuildingPositions = function()
+	return {}
+end
+local IsMexPositionValid = function()
+	return false
+end
+
 local unitSizesQuad = {}
 for uDefID, uDef in pairs(UnitDefs) do
 	if uDef.extractsMetal > 0 then
-		isMex[uDefID] = uDef.extractsMetal * 1000
-		unitSizesQuad[uDefID] = {uDef.xsize*4, uDef.zsize*4}
-	end
-end
-local isMexConstructor = {}
-for uDefID, uDef in pairs(UnitDefs) do
-	if uDef.buildOptions then
-		for _, option in ipairs(uDef.buildOptions) do
-			if isMex[option] then
-				isMexConstructor[uDefID] = uDef.buildOptions
-				break
-			end
-		end
+		unitSizesQuad[uDefID] = { uDef.xsize * 4, uDef.zsize * 4 }
 	end
 end
 
 local function GetExtractionAmount(spot, metalExtracts, orders)
-	local spotWorth = spot.worth/1000.0
```

Comment:
I assume finder was changed to already do this inside of it, no idea why this was necessary before and not anymore, didn't care to dig what introduced this regression.